src/index.ts

```diff
@@ -44,43 +42,62 @@ async function main() {
-
-  // Output total results and transaction files for input to next process, details file for audit log
-  console.log(`We won these blocks: ${blocksIncluded}`);
-  console.log(`We are paying out based on total rewards of ${allBlocksTotalRewards} nanomina in this window.`);
-  console.log(`That is ${allBlocksTotalRewards / 1000000000} mina`);
-  console.log(`The Pool Fee is is ${allBlocksTotalPoolFees / 1000000000} mina`);
-  console.log(`Total Payout should be ${(allBlocksTotalRewards) - (allBlocksTotalPoolFees)} nanomina or ${((allBlocksTotalRewards) - (allBlocksTotalPoolFees)) / 1000000000} mina`)
-  console.log(`The Total Payout is actually: ${totalPayout} nm or ${totalPayout / 1000000000} mina`)
-
-  const runDateTime = new Date();
-  const payoutTransactionsFileName = generateOutputFileName("payout_transactions", runDateTime, minimumHeight, maximumHeight);
-
-  fs.writeFile(payoutTransactionsFileName, JSON.stringify(payouts), function (err: any) {
-    if (err) throw err;
-    console.log(`wrote payouts transactions to ${payoutTransactionsFileName}`);
+  let payouts: PayoutTransaction[] = [];
+  let storePayout: PayoutDetails[] = [];
+  const ledgerHashes = [...new Set(blocks.map(block=>block.stakingledgerhash))];
+  Promise.all(ledgerHashes.map(async ledgerHash => {
+    console.log(`### Calculating payouts for ledger ${ledgerHash}`)
+    const [stakers, totalStake] = getStakes(ledgerHash, stakingPoolPublicKey, globalSlotStart, slotsPerEpoch);
+    console.log(`The pool total staking balance is ${totalStake}`);
+    
+    // run the payout calculation for those blocks
+    const ledgerBlocks = blocks.filter(x => x.stakingledgerhash == ledgerHash);
+    const [ledgerPayouts, ledgerStorePayout, blocksIncluded, allBlocksTotalRewards, allBlocksTotalPoolFees, totalPayout] = await getPayouts(ledgerBlocks, stakers, totalStake, commissionRate, transactionFee);
+    payouts.push(...ledgerPayouts);
+    storePayout.push(...ledgerStorePayout);
+
+    // Output total results and transaction files for input to next process, details file for audit log
+    console.log(`We won these blocks: ${blocksIncluded}`);
+    console.log(`We are paying out based on total rewards of ${allBlocksTotalRewards} nanomina in this window.`);
+    console.log(`That is ${allBlocksTotalRewards / 1000000000} mina`);
+    console.log(`The Pool Fee is ${allBlocksTotalPoolFees / 1000000000} mina`);
+    console.log(`Total Payout should be ${(allBlocksTotalRewards) - (allBlocksTotalPoolFees)} nanomina or ${((allBlocksTotalRewards) - (allBlocksTotalPoolFees)) / 1000000000} mina`)
+    console.log(`The Total Payout is actually: ${totalPayout} nm or ${totalPayout / 1000000000} mina`)
+  })).then(()=>{
+    // Aggregate to a single transaction per key
+    const transactions = [...payouts.reduce((r, o) => {
+      const item = r.get(o.publicKey) || Object.assign({}, o, {
+        amount: 0,
+        fee: 0
+      });
+      item.amount += o.amount;
+      item.fee += o.fee;
```

Comment:
We want to pay the minimum fee and should not aggregate it. use transactionFee that's set from the env; although, what we really need to be doing is calling a function to calculate a recommended fee, and then decide whether the pool pays the fee, or whether the staker pays the fee. Suggest we stub this for now with a call to getRecommendedFee -- probably will need it's own file because it will need to get data from somewhere too. Maybe too complicated for this PR -- let's leave it at not adding extra fees and then address this when we handle Issue 21.